packages/clarity-js/src/layout/region.ts

```diff
@@ -42,23 +44,39 @@ export function track(id: number): void {
 }
 
 export function compute(): void {
+    // Process any regions that couldn't be processed earlier
+    let q = [];
+    for (let r of queue) {
+        let id = dom.getId(r.node);
+        if (id) {
+            r.data.id = id;
+            data.push(r.data);
+        } else { q.push(r); }
```

Comment:
This looks like it could get stuck in an ever growing queue if things did not work out. Perhaps limit the number of retries or  timeout them out.

Reply:
Discussed this offline and we are in sync. This will be capped at number of the regions and I've added more comments to reflect the intent of this code path.